database-plugins/src/main/java/io/cdap/plugin/db/connector/DBConnector.java

```diff
@@ -146,130 +95,23 @@ public BrowseDetail browse(ConnectorContext context, BrowseRequest browseRequest
     });
   }
 
-  @Override
-  public ConnectorSpec generateSpec(ConnectorContext context, ConnectorSpecRequest connectorSpecRequest)
-    throws IOException {
-
-    return connectAndQuery(connectorSpecRequest.getPath(), "generate spec", (connection, path) -> {
-      ConnectorSpec.Builder specBuilder = ConnectorSpec.builder();
-      DatabaseMetaData metaData = connection.getMetaData();
-      Map<String, String> properties = new HashMap<>();
-      properties.put(DBBaseConfig.NAME_USE_CONNECTION, "true");
-      properties.put(DBBaseConfig.NAME_CONNECTION, connectorSpecRequest.getConnectionWithMacro());
-      String table = path.getTable();
-      if (table != null) {
-        String schema = path.getSchema();
-        ResultSet columns = metaData.getColumns(connection.getCatalog(), schema, table, null);
-        List<Schema.Field> fields = new ArrayList<>();
-        while (columns.next()) {
-          int sqlType = columns.getInt(RESULTSET_COLUMN_DATA_TYPE);
-          String typeName = columns.getString(RESULTSET_COLUMN_TYPE_NAME);
-          int scale = columns.getInt(RESULTSET_COLUMN_DECIMAL_DIGITS);
-          int precision = columns.getInt(RESULTSET_COLUMN_COLUMN_SIZE);
-          String columnName = columns.getString(RESULTSET_COLUMN_COLUMN_NAME);
-          Schema columnSchema = DBUtils.getSchema(typeName, sqlType, precision, scale, columnName, true);
-          String isNullable = columns.getString(RESULTSET_COLUMN_IS_NULLABLE);
-          if ("YES".equals(isNullable)) {
-            columnSchema = Schema.nullableOf(columnSchema);
-          }
-          fields.add(Schema.Field.of(columnName, columnSchema));
-        }
-        specBuilder.setSchema(Schema.recordOf("output", fields));
-        properties.put(DBSource.DBSourceConfig.IMPORT_QUERY,
-                       schema == null ? String.format("SELECT * FROM %s;", table) :
-                         String.format("SELECT * FROM %s.%s;", schema, table));
-        properties.put(DBSource.DBSourceConfig.NUM_SPLITS, "1");
-      }
-      return specBuilder.addRelatedPlugin(new PluginSpec(DBSource.NAME, BatchSource.PLUGIN_TYPE, properties)).build();
-    });
-  }
+  protected void setConnectorSpec(ConnectorSpecRequest request, DBConnectorPath path,
```

Comment:
rename it to addRelatedPlugins ?

Reply:
You can use it to set whatever you want, properties, even the schema, so I name it like this